R/integrate_sces.R

```diff
@@ -0,0 +1,180 @@
+# This function can perform the crux integration based on the given integration method
+
+
+#' Integrate a merged set of SingleCellExperiment objects using a specified
+#'  integration method. The final SCE object will contain an additional reducedDim
+#'  entitled `{integration_method}_PCA`. All original SCE assays are retained.
+#'  `fastMNN` integration uses all default setting, but additional parameters can be supplied.
+#'  `harmony` integration uses existing PCs, but additional parameters can be supplied.
+#'
+#' @param merged_sce A merged SCE object as prepared by `scpcaTools::merge_sce_list()`.
+#' @param integration_method The integration method to use. One of `fastMNN` or
+#'  `harmony` (case-sensitive)
+#' @param batch_column The column in the merged SCE object indicating batches
+#' @param covariate_cols A vector of additional columns in the merged SCE to
+#'   consider as covariates during integration. Currently, this is used only by
+#'   `harmony`.
+#' @param return_corrected_expression A boolean indicating whether corrected expression
+#'   values determined by the given integration method should be included in the
+#'   integrated SCE object. Note that `harmony` does not calculate corrected expression,
+#'   so this argument is ignored for this integration method. Default is `TRUE`.
+#' @param seed Random seed to set for integration. This is only set if the value is not `NULL`.
+#' @param ... Any additional parameters to be passed to the given integration method
+#'
+#' @return An updated merged SCE object containing a new reduced dimension named
+#'   `{integration_method}_PCA` containing the corrected PCs. In the case of `fastMNN`
+#'   integration, the SCE also includes a corrected expression assay (`fastMNN_corrected`)
+#'
+#' @import SingleCellExperiment
+#'
+#' @export
+integrate_sces <- function(merged_sce,
+                           integration_method = c("fastMNN", "harmony"),
+                           batch_column = "sample",
+                           covariate_cols = c(),
+                           return_corrected_expression = TRUE,
```

Comment:
Wondering if this is the default we want. I think `FALSE` maybe makes more sense (not sure how often we will use this), and then add a warning for `harmony` if true since this is not an option there.